core/filestorage/filesync/upload.go

```diff
@@ -227,17 +243,54 @@ func (f *fileSync) prepareToUpload(ctx context.Context, spaceId string, fileId s
 		)
 	}
 
-	stat, err := f.getAndUpdateSpaceStat(ctx, spaceId)
+	if len(blocksToUpload) > 0 && bytesToUpload > vacantSpace {
+		return nil, errReachedLimit
+	}
+
+	return blocksToUpload, nil
+}
+
+// estimateFileSize use heuristic to estimate file size. It's pretty accurate for ordinary files,
+// But getting worse for images because they have a different structure.
+// Samples of estimation errors in bytes (the bigger the file, the bigger the error):
+/*
+	Estimation error: 968 — 10Mb jpeg file
+	Estimation error: 1401 — 30Mb png file
+	Estimation error: 810 — 50Mb ordinary file
+	Estimation error: 3576 — 250Mb ordinary file
+*/
+// Ordinary file structure:
+/*
+- dir (root)
+  	- dir (content and meta pair)
+  		- meta
+  		- content (just content if file < 1Mb, and chunks list otherwise)
+			- chunk1
+			- chunk2
+			...
+*/
+func (f *fileSync) estimateFileSize(fileID string) (int, error) {
```

Comment:
Maybe to add tests for this functionality, if it's possible?

Reply:
It's too hard to write tests for this behavior properly. It's better to invest time on writing tests for `core/files` package. And then comes Files as objects project where we will fix the reason for this size estimation